criu/img-remote.c

```diff
@@ -940,11 +940,20 @@ static int64_t send_image_async(struct roperation *op)
 	struct rimage *rimg = op->rimg;
 	bool close_fd = op->close_fd;
 	int n;
+	int flags = 0;
+
+#ifdef MSG_ZEROCOPY
+	flags |= MSG_ZEROCOPY;
+#endif
 
-	n = write(
+	n = send(
 		fd,
 		op->curr_sent_buf->buffer + op->curr_sent_bytes,
-		min(BUF_SIZE, op->curr_sent_buf->nbytes) - op->curr_sent_bytes);
+		min(BUF_SIZE, op->curr_sent_buf->nbytes) - op->curr_sent_bytes,
+		flags);
+
```

Comment:
@rst0git , are you sure it's worth the effort?

The kernel docs say: "MSG_ZEROCOPY is generally only effective at writes over around 10 KB" and much of the transfers here would be way less than 10 KB...

Reply:
That is a good point, here we only send the image header which has very small size.